api/v1/tenant_types.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023 Tigera, Inc. All rights reserved.
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TenantSpec struct {
+	// ID is the unique identifier for this tenant.
+	// +required
+	ID string `json:"id,omitempty"`
+}
+
+type TenantStatus struct{}
```

Comment:
We should probably have a detailed discussion for status handling here before I implement anything too far. There are a few ways we could go:

- Namespaced TigeraStatus objects, inside the tenant's namespace.
- Global TigeraStatus objects, no "tenant" TS, but adding status to the Tenant object itself
- Just don't worry about status for the Tenant API since it's covered elsewhere (I think?)

Reply:
I'm totally on board with having a discussion on the status handling.